### DISCUSSION TITLE
QInput - anyway to transform the displayed value?

### DISCUSSION THREAD

**Discussion by yeechutngeow** [quasarframework/quasar#15973] [General - Components / Directives / etc]
I am looking for a q-input to input time duration with the display in hours:minutes format, something like the following:

`  <q-input type="number" v-model="duration" step="15" max="120" min="30" label="Duration (hh:mm)" />`

but the display values should be like "0:15", "0:30", "0:45", "1:00", "1:15", "1:30", "2:00".
I thought the default slot is for such purposes - but the slot adds an extra text at the top right corner, which seems completely illogical, and it makes me wonder whether there is a bug here.

**Comment by suleimanas** [comment]
the slots has nothing to do with the value itself, what you want is the mask prop: `mask="time"`
`<q-input type="number" v-model="duration" label="Duration (hh:mm)" mask="time" />`

you can also use `type="time"`
`<q-input type="time" v-model="duration" label="Duration (hh:mm)" />`

**Comment by yeechutngeow** [comment]
Type="time" is for specifying the time of day, not duration in hours and minutes. Also with type="number" I have the spin up and spin down button and the increment/decrement step.
